### ISSUE TITLE
FR: support for aliascnt package

### ISSUE THREAD

**Issue by mbertucci47** [brucemiller/LaTeXML#2444]
This is a feature request for support for the [aliascnt](https://www.ctan.org/pkg/aliascnt) package. My main use case for it is creating sibling theorems that work correctly with cleveref and hyperref's `\autoref`. Here's an example.
```tex
\documentclass{article}
\usepackage{amsthm}
\usepackage{hyperref}
\usepackage{cleveref}

\newtheorem{theorem}{Theorem}
\newtheorem{lemma}[theorem]{Lemma}

\providecommand{\lemmaautorefname}{Lemma}

\begin{document}

\cref{foo}

\autoref{foo}

\begin{lemma}\label{foo}
Some lemma text
\end{lemma}

\end{document}
```
![test-1](https://github.com/user-attachments/assets/579cf541-cf89-437a-8350-1158d81255b9)

So the lemma is incorrectly referred to as a theorem by `\cref` and `\autoref`. But with the features provided by aliascnt, the names are correct.
```tex
\documentclass{article}
\usepackage{amsthm}
\usepackage{aliascnt}
\usepackage{hyperref}
\usepackage{cleveref}

\newtheorem{theorem}{Theorem}
\newaliascnt{lemma}{theorem}
\newtheorem{lemma}[lemma]{Lemma}
\aliascntresetthe{lemma}

\providecommand{\lemmaautorefname}{Lemma}

\begin{document}

\cref{foo}

\autoref{foo}

\begin{lemma}\label{foo}
Some lemma text
\end{lemma}

\end{document}
```
![test-1](https://github.com/user-attachments/assets/a8a87750-7e73-4a60-89d9-ef8853ec077a)

**Comment by xworld21** [comment]
Strange, I cannot reproduce your example. Cleveref does not need aliascnt to work, only hyperref does. I get 'lemma 1 / Theorem 1' with LaTeX (with many versions of TeX Live) and 'lemma 1 / Lemma 1' with LaTeXML *without* using aliascnt. If anything, LaTeXML's implementation of `\autoref` is not faithful. Which versions are you using, exactly?

LaTeXML needs additional calls to `\crefname` when using nonstandard theorem environments (already reported). Here 'theorem' and 'lemma' work out of the box.

**Comment by mbertucci47** [comment]
@xworld21 The cleveref issue needs a current LaTeX (2024-11-01 or later) to reproduce. And you're right about LaTeXML not being faithful with `\autoref`.

I now see the issue you're referring to, #2306.

**Comment by xworld21** [comment]
> The cleveref issue needs a current LaTeX (2024-11-01 or later) to reproduce. And you're right about LaTeXML not being faithful with `\autoref`.

Oh no, that sounds like a LaTeX regression. I suggest reaching out to the cleveref developer and the TeX Live mailing list because it sounds like an unintended breaking change (I am happy to do it if you prefer – I am a heavy cleveref user myself!).

> By "nonstandard", do you mean using amsthm instead of the kernel `\newtheorem`?

I confess I don't actually have a definition of 'standard'. I am referring to the environment names 'theorem', 'lemma', etc. Many are automatically supported by LaTeXML, although I do not remember how, exactly. If you define environments like 'thm', 'lem', you will need to call `\crefname` by hand.

**Comment by mbertucci47** [comment]
@xworld21 The cleveref maintainer has been unresponsive for years and the LaTeX team is well-aware of the issues with recent kernels and cleveref. They are constantly patching the cleveref code in firstaid because cleveref makes assumptions about internal kernel code that no longer apply. See https://github.com/latex3/latex2e/issues/1393, https://github.com/latex3/hyperref/issues/362, https://tex.stackexchange.com/q/730148/208544, https://tex.stackexchange.com/q/711418/208544; really just search "cleveref" on TeX-SX to see the long list of issues. The recommendation is to move to [zref-clever](https://www.ctan.org/pkg/zref-clever) but obviously it will take a long time to convince most people to do this.

**Comment by xworld21** [comment]
> The recommendation is to move to [zref-clever](https://www.ctan.org/pkg/zref-clever) but obviously it will take a long time to convince most people to do this.

Oh. I suppose LaTeXML should support that too.

**Comment by mbertucci47** [comment]
For the record, the issue explained in https://github.com/brucemiller/LaTeXML/issues/2444#issue-2737481192 is fixed in the June 2026 LaTeX release, so aliascnt is no longer necessary.

**Comment by brucemiller** [comment]
I'm a bit confused here, but I think I understand that *both* references should name it a "lemma" (or specifically "lemma 1" and "Lemma 1"). Which, in the meantime, is exactly what the current LaTeXML produces.  (The cruel irony is that my local TeX-live produces lemma and Theorem, which is wrong, right? :> )

So, is this closable, or is `aliascnt` still really needed?

**Comment by xworld21** [comment]
I think `aliascnt` is still worth implementing (if somebody can find the time!), people may still be using it for other reasons.

**Comment by mbertucci47** [comment]
@brucemiller Yes it should be "lemma 1" and "Lemma 1". With or without aliascnt, this is produced with a current (2026) texlive, so the original use case is no longer relevant. However, as @xworld21 mentions, the package could be used for things other than theorems.
